src/recorder.h

```diff
@@ -46,16 +46,20 @@ class Recorder {
   friend class Datum;
 
  public:
-  /// create a new recorder with default dump frequency and random
-  /// simulation id.
+  /// create a new recorder with default dump frequency, random
+  /// simulation id, and simulation id injection.
   Recorder();
 
+  /// create a new recorder with the given dump count, random
+  /// simulation id, and given flag for injecting the simulation id.
+  Recorder(bool inject_sim_id);
+
   /// create a new recorder with the given dump count and random
   /// simulation id.
   Recorder(unsigned int dump_count);
 
-  /// create a new recorder with default dump frequency and the specified
-  /// simulation id.
+  /// create a new recorder with default dump frequency. the specified
+  /// simulation id, and simulation id injection.
   Recorder(boost::uuids::uuid simid);
```

Comment:
@scopatz, you can use default arguments [in constructors](http://google-styleguide.googlecode.com/svn/trunk/cppguide.html#Default_Arguments). I think some slight retooling here (as you see fit) would be nice, rather than having on constructor per argument. Also, I think it would be nice to have a constructor that takes all arguments (non default).